### PR TITLE
How to setup ci/cd to react with circle ci, aws s3, and aws cloud front

### LINKED ISSUE #668: How to setup CI/CD to React with Circle CI, AWS S3, and AWS CloudFront
### Brief Summary:
First, we are going to set up the GitHub repository. After the successful setup of the GitHub repo, we need to register to an AWS account and navigate to the console. We are going to setup AWS S3 with all the permissions and local deployment configured. Then, move on to setting up CircleCI using a GitHub account. Lastly, we use AWS CloudFront to deploy our project.

### Key Takeaways:
After completing this tutorial article, you will be able to
- To set up the Git version tool along with GitHub.
- To set up AWS S3 service.
- To set up CircleCI.
- To set up AWS CloudFront service.

### References:
- [What is CI & CD](https://www.infoworld.com/article/3271126/what-is-cicd-continuous-integration-and-continuous-delivery-explained.html)
- [Continuous Integration. CircleCI vs Travis CI vs Jenkins](https://hackernoon.com/continuous-integration-circleci-vs-travis-ci-vs-jenkins-41a1c2bd95f5)
- [Speed Up Your npm Workflow With npx](https://alligator.io/workflow/npx/)
- [How we used CircleCI 2.0 to build and deploy an Angular app to AWS S3](https://medium.freecodecamp.org/our-journey-for-using-circleci-2-0-to-build-and-deploy-an-angular-app-to-aws-s3-8e7ea3f51503)
- [React with CircleCI, AWS S3, and AWS CloudFront](https://blog.cloudboost.io/react-with-circleci-aws-s3-and-aws-cloudfront-844a1b2c75c9)
- [Automating with CircleCi 2.0](https://medium.com/@RockChalkDev/automating-with-circleci-2-0-57de968bda90)

### PR DESCRIPTION
**Pull request by Worawat101**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
This fixes #668 
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

File: articles/ci-cd-setup-react/index.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+![In Progress](21-in-progress-status.png)
+
+After the status changes to **Deployed**, we will be able to access the site from the domain name [https://dhhqbyhy7gv01.cloudfront.net/](https://dhhqbyhy7gv01.cloudfront.net/). Here, our transmission protocol is already set to HTTPS. The access site from the above domain name is shown in the demo screenshot below:
+
+![final result](22-final-result.png)
+
+Hence, we have successfully set up the CircleCI and AWS CloudFront to our React Native project.
+
+### What’s next …
```
Comment (ninjaginja) [inline articles/ci-cd-setup-react/index.md:244]: suggest leaving this out until the content becomes available

**Review by ninjaginja — CHANGES_REQUESTED** [review]
Good article @Worawat101. I've made some minor formatting and readability edits. Please see requested edits for the beginning of the article.

**Review by Worawat101 — COMMENTED** [review]
@ninjaginja updated following your suggestion.

**Review by ninjaginja — APPROVED** [review]

**Comment by ninjaginja** [comment]
I made some edits to the intro to improve readability. Please take a look and note for future articles. Thanks for your contribution - great article!

**Pull request merged**